### PR TITLE
Count changes for each store

### PR DESCRIPTION
**Pull request by cymruu**
A proposition to resolve issue #1134

### CODE REVIEW

File: pkg/action/sync.go [hunk 1]
```diff
@@ -42,29 +37,44 @@ func (s *Action) sync(ctx context.Context, c *cli.Context, store string) error {
...
+	if len(changes) > 0 {
+		out.Print(ctx, "Sync summary:\n")
+	}
+	for k := range changes {
+		if changes[k] > 0 {
+			out.Print(ctx, fmt.Sprintf("Mount %s: Added %d entries", k, changes[k]))
+			totalAdded += changes[k]
+		} else if changes[k] < 0 {
+			out.Print(ctx, fmt.Sprintf("Mount %s: Removed %d entries", k, -1*changes[k]))
+			totalRemoved += -1 * changes[k]
```
Comment (dominikschulz) [inline pkg/action/sync.go]: Remove the ´-1 *´, just add up what's removed.

File: pkg/action/sync.go [hunk 2]
```diff
@@ -42,29 +37,44 @@ func (s *Action) sync(ctx context.Context, c *cli.Context, store string) error {
...
+		} else {
+			out.Print(ctx, fmt.Sprintf("Mount %s: no changes", k))
+		}
 	}
 	diff := ""
```
Comment (dominikschulz) [inline pkg/action/sync.go]: Maybe use a strings.Builder here?

File: pkg/action/sync.go [hunk 3]
```diff
@@ -42,29 +37,44 @@ func (s *Action) sync(ctx context.Context, c *cli.Context, store string) error {
...
-	if numEntries > 0 {
-		diff = fmt.Sprintf(" Added %d entries", numEntries)
-	} else if numEntries < 0 {
-		diff = fmt.Sprintf(" Removed %d entries", -1*numEntries)
+	if totalAdded > 0 {
+		diff += fmt.Sprintf(" Added %d entries", totalAdded)
+	}
+	if totalRemoved > 0 {
+		diff += fmt.Sprintf("\nRemoved %d entries", totalRemoved)
```
Comment (dominikschulz) [inline pkg/action/sync.go]: Remove the "\n", please.

File: pkg/action/sync.go [hunk 4]
```diff
@@ -42,29 +37,44 @@ func (s *Action) sync(ctx context.Context, c *cli.Context, store string) error {
...
 	}
 	_ = notify.Notify(ctx, "gopass - sync", fmt.Sprintf("Finished. Synced %d remotes.%s", numMPs, diff))
 
 	return nil
 }
 
 // syncMount syncs a single mount
-func (s *Action) syncMount(ctx context.Context, mp string) error {
+func (s *Action) syncMount(ctx context.Context, mp string) (*syncResult, error) {
```
Comment (dominikschulz) [inline pkg/action/sync.go]: Don't use a pointer to syncResult. We don't need to modify the object later.

File: pkg/action/sync.go [hunk 5]
```diff
@@ -90,24 +100,25 @@ func (s *Action) syncMount(ctx context.Context, mp string) error {
...
+			return nil, err
 		}
 
 		out.Error(ctx, "Failed to push '%s' to its remote: %s", name, err)
-		return err
+		return nil, err
 	}
 	out.Print(ctxno, color.GreenString("OK"))
-
+	syncResult := &syncResult{name, 0}
```
Comment (dominikschulz) [inline pkg/action/sync.go]: Don't explicitly init default values, i.e. drop the 0.

File: pkg/action/sync.go (hunk 4 above)
Comment (cymruu) [reply pkg/action/sync.go]: Isn't that convenient so I can return nil?

File: pkg/action/sync.go (hunk 1 above)
Comment (cymruu) [reply pkg/action/sync.go]: Since if entries were removed it returns negative number it's necessary to not print "removed -5 entries". Or I am wrong?

**Pull request closed without merging**